ani-cli

```diff
@@ -117,103 +125,85 @@ search_eps () {
 search_for_unwatched () {
 	# compares history with gogoanime, only shows unfinished anime
 	search_results=$*
-
-	unwatched_anime=""
+	unwatched_anime=''
 	while read -r anime_id; do
 		current_ep_number="$(search_eps "$anime_id")"
 		history_ep_number="$(sed -n -E "s/${anime_id}\t//p" "$logfile")"
-		if [ "$current_ep_number" -ge "$history_ep_number" ]
-		then
-			# breaks down with \n because of the display functions
-			unwatched_anime="$unwatched_anime$anime_id
-"
+		if [ "$current_ep_number" -ge "$history_ep_number" ]; then
+			unwatched_anime='1'
+			echo "$anime_id"
 		fi
 	done <<-EOF
 	$search_results
 	EOF
-	if [ -z "$unwatched_anime" ]; then
-		die "No unwatched episodes"
-	fi
-	printf "%s" "$unwatched_anime"
+	[ -z "$unwatched_anime" ] && die 'No unwatched episodes'
 }
 
 search_another_anime () {
 	tput clear
-	prompt "Search Anime"
+	prompt 'Search Anime'
 	query="$REPLY $REPLY2"
-	search_results=$(search_anime "$query")
-	[ -z "$search_results" ] && die "No search results found"
-	anime_selection "$search_results"
-	episode_selection
+	process_search
 }
 
 search_history () {
-	tput clear
-	search_results=$(sed -n -E 's/\t[0-9]*//p' "$logfile")
-	[ -z "$search_results" ] && die "History is empty"
-	search_results=$(search_for_unwatched "$search_results")
+	search_results="$(sed -n -E 's/\t[0-9]*//p' "$logfile")"
```

Comment:
why are you adding quotes here?
Shell word splitting is not applicable here. Word splitting does not occur in variable assignment.
source: posix sh specs 2.9.1 point 2

Reply:
It was more of a why not then why for me, but I can remove it if you wish

Reply:
Check out https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/, this is the specifications of shell scripting. Shell is beautiful, it has a grammar and then syntax that dictates what happens when.